src/shared/utils/file-operations.js

```diff
@@ -62,3 +73,42 @@ export function selectAndUploadFile(notebookID, successCallback = () => {}) {
       .then(output => successCallback(output));
   });
 }
+
+export function makeFormData(notebookID, data, fileName) {
+  const file = valueToFile(data, fileName);
+  const formData = new FormData();
+  formData.append(
+    "metadata",
+    JSON.stringify({
+      filename: file.name,
+      notebook_id: notebookID
+    })
+  );
+  formData.append("file", file);
+  return formData;
+}
+
+export function saveFileToServer(
+  notebookID,
+  data,
+  fileName,
+  fileID = undefined
+) {
+  const formData = makeFormData(notebookID, data, fileName);
+  const fetchRequest = fileID // if fileID undefined, upload, otherwise update
+    ? fd => updateFile(fileID, fd)
+    : fd => uploadFile(fd);
```

Comment:
let's get rid of the `uploadFile` function altogether -- this logic can be handled within `updateFile` based on whether or not an fileId is passed in

Reply:
hmm, might leave this as-is for now and file a follow-on PR. This is from the PR this is rebased on, not this PR itself, and @openjck is building off of this at the moment. For now I'd prefer to just leave a `FIXME` in that PR since it is not merged yet at any rate. I agree we should clean up this code but I think it'd be a good idea not to shake things up too much while a bunch of stuff depends on this.

Reply:
yep makes sense :+1:

Reply:
...but please file a code quality issue linking back to this discussion before merging this